index.js

```diff
@@ -7,14 +7,24 @@ export default function createBroadcast (initialState) {
 
   const setState = state => {
     _state = state
-    Object.keys(listeners).forEach(id => listeners[id](_state))
+    const keys = Object.keys(listeners)
+    for (let i = 0; i < keys.length; i += 1) {
+      // if a listener gets unsubscribed during setState we just skip it
+      if (typeof listeners[keys[i]] !== 'undefined') {
```

Comment:
I've added a check to throw an error in case you do not pass a function to subscribe